packages/components/src/post-locked-modal/index.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Modal } from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	PostPreviewButton,
+} from '@wordpress/editor';
+import { withSelect } from '@wordpress/data';
+
+import './style.scss';
+import { addQueryArgs } from '@wordpress/url';
+import { getPostEditURL } from '../../../../edit-post/components/browser-url';
+import { getWPAdminURL } from '../../../editor/src/utils/url';
+
+class PostLockedModal extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			isOpen: true,
+		};
+
+		this.takeOverPost = this.takeOverPost.bind( this );
+		const { lockDetails } = this.props;
+
+		if ( lockDetails && lockDetails.text ) {
+			this.modalText = lockDetails.text;
+			this.avatar = lockDetails.avatar_src;
+			this.takeover = true;
+		} else {
+			const { user } = this.props;
+			const displayName = ( user && user.data ) ? user.data.display_name : __( 'Another user' );
+			this.modalText = sprintf( __( '%s is already editing this post. Do you want to take over?' ), displayName );
+			this.avatar = user.data.avatar_src;
+		}
+	}
+
+	takeOverPost() {
+		const { lockNonce, id } = this.props();
+		const unlockUrl = addQueryArgs( getPostEditURL(), {
+			'get-post-lock': '1',
+			_wpnonce: lockNonce,
+			lockKey: true,
+			post: id,
+		} );
+		document.location = unlockUrl;
```

Comment:
Why do we assign these behaviors as button and not just use a standard anchor tag with `href` ?

Reply:
Probably because the code originated int he core version, I can update this to a standard href tag.

Reply:
I switched this to standard anchor tags in https://github.com/WordPress/gutenberg/pull/4217/commits/b7ff092f3bd8acba221eed4838d3067a5687326e. i found it useful to keep the button components for correct styling.